packages/test/test-end-to-end-tests/src/test/waitForSummary.spec.ts

```diff
@@ -0,0 +1,65 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { describeCompat } from "@fluid-private/test-version-utils";
+import type { IContainer } from "@fluidframework/container-definitions/internal";
+import type { ISharedMap } from "@fluidframework/map/internal";
+import {
+	type ITestObjectProvider,
+	type ITestContainerConfig,
+	createSummarizer,
+	summarizeNow,
+	type ChannelFactoryRegistry,
+	createAndAttachContainer,
+	DataObjectFactoryType,
+	type ITestFluidObject,
+} from "@fluidframework/test-utils/internal";
+
+const waitForSummary = async (
+	provider: ITestObjectProvider,
+	container: IContainer,
+	testContainerConfig: ITestContainerConfig,
+) => {
+	const { summarizer, container: summarizingContainer } = await createSummarizer(
+		provider,
+		container,
+		testContainerConfig,
+	);
+	await summarizeNow(summarizer);
+	summarizingContainer.close();
+};
+
+describeCompat("Wait for summary", "NoCompat", (getTestObjectProvider, apis) => {
+	const mapId = "map";
+	const { SharedMap } = apis.dds;
+	const registry: ChannelFactoryRegistry = [[mapId, SharedMap.getFactory()]];
+
+	it("Wait for summary", async function () {
+		const provider = getTestObjectProvider();
+		// This test fails on ODSP at least.
+		if (provider.driver.type !== "local") {
+			this.skip();
+		}
+		const testContainerConfig: ITestContainerConfig = {
```

Comment:
This config doesn't disable summaries. So, there are chances of a race condition where this container may spawn a summarizer client. This would then compete with the summarizer client that is explicitly created in `waitForSummary`

Reply:
I'll disable it again for main container.